### PR TITLE
policy: Add template policy

### PR DESCRIPTION
**Pull request by oxxenix**
This is an example policy that can be used to develop a new policy with proper functionality.

### CODE REVIEW

File: cmd/template/main.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	"time"
+
+	"github.com/intel/nri-resmgr/pkg/config"
+	"github.com/intel/nri-resmgr/pkg/instrumentation"
+	"github.com/intel/nri-resmgr/pkg/resmgr"
+
+	logger "github.com/intel/nri-resmgr/pkg/log"
+	version "github.com/intel/nri-resmgr/pkg/version"
+
+	_ "github.com/intel/nri-resmgr/cmd/template/policy"
```
Comment (klihub) [inline cmd/template/main.go:31]: These import paths need to be updated to refer to this `github.com/containers/nri-plugins` repo.

File: cmd/template/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# Build nri-resmgr
+COPY . .
+
+RUN make clean
+RUN make build-static
+
+FROM gcr.io/distroless/static
+
+COPY --from=builder /go/builder/build/bin/nri-resmgr-template /bin/nri-resmgr-template
```
Comment (jukkar) [inline cmd/template/Dockerfile:20]: please change `resmgr` to `resource-policy` here and other places in this PR

File: cmd/template/fallback.cfg.sample [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# stored in the cache).
+#
+
+policy:
+  Active: template
+  ReservedResources:
+    CPU: 750m
+logger:
+  Debug: resource-manager,cache,policy,resource-control
+
```
Comment (jukkar) [inline cmd/template/fallback.cfg.sample:14]: There is now extra line at the end of the file, please remove it.

File: cmd/template/main.go (hunk 1 above)
Comment (jukkar) [inline cmd/template/main.go:31]: The module location has changed, please change
`intel/nri-resmgr/`
to
`containers/nri-plugins`

File: cmd/template/nri-resmgr-template-deployment.yaml.in [hunk 4]
```diff
@@ -0,0 +1,142 @@
+apiVersion: v1
```
Comment (jukkar) [inline cmd/template/nri-resmgr-template-deployment.yaml.in:1]: rename the file to `nri-resource-policy-template-deployment.yaml.in`

File: cmd/template/nri-resmgr-template-deployment.yaml.in [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+kind: ServiceAccount
+metadata:
+  name: nri-resmgr
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: nri-resmgr
```
Comment (jukkar) [inline cmd/template/nri-resmgr-template-deployment.yaml.in:10]: rename all `nri-resmgr` to `nri-resource-policy` in this file

File: cmd/template/nri-resmgr-template-deployment.yaml.in [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      serviceAccount: nri-resmgr
+      nodeSelector:
+        kubernetes.io/os: "linux"
+      containers:
+        - name: nri-resmgr-template
+          args:
+            - --host-root
+            - /host
+            - --fallback-config
+            - /etc/nri-resmgr/nri-resmgr.cfg
```
Comment (jukkar) [inline cmd/template/nri-resmgr-template-deployment.yaml.in:74]: `nri-resmgr` directory is changed to `nri-resource-policy` and filename to `nri-resource-policy.cfg`

**Review by jukkar — CHANGES_REQUESTED** [review]
Generally lgtm, but please do search and replace all mentions of
nri-resmgr to nri-resource-policy in this PR.

File: cmd/template/policy/template-policy.go [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+package template
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/intel/nri-resmgr/pkg/cache"
+	"github.com/intel/nri-resmgr/pkg/introspect"
+	logger "github.com/intel/nri-resmgr/pkg/log"
+	policyapi "github.com/intel/nri-resmgr/pkg/policy"
+	"github.com/intel/nri-resmgr/pkg/resmgr/events"
```
Comment (klihub) [inline cmd/template/policy/template-policy.go:24]: These import paths need to be updated to point to the new location.

**Comment by jukkar** [comment]
> Everything is now updated. If correct, could you please squash commits when merging?

You can do it yourself. Just squash them together `git rebase -i HEAD~2` and then force push to this branch.

File: cmd/template/fallback.cfg.sample [hunk 8]
```diff
@@ -10,5 +10,4 @@ policy:
   ReservedResources:
     CPU: 750m
 logger:
-  Debug: resource-manager,cache,policy,resource-control
-  
\ No newline at end of file
+  Debug: resource-manager,cache,policy,resource-control
```
Comment (jukkar) [inline cmd/template/fallback.cfg.sample:13]: There is something wrong with this file, github complains that "no newline at the of file", can you verify that there is a newline "\n" at the end.

Comment (jukkar) [reply cmd/template/fallback.cfg.sample:13]: Probably that is related to your editor then. You can see the exact characters what the commit contains with this command:
```
git show HEAD~1 | od -A x -t x1z -v -c
```
The last character of the file should be \n in this case, and not an empty line (which means two \n).

**Pull request closed without merging**